### PR TITLE
Adjust SMTChecker timeout tests

### PR DESCRIPTION
**Pull request by leonardoalt**
Timeouts are very prone to nondeterminism, especially since we have at least 4 different CI environments.
These tests need to be hard enough so both BMC and CHC fail after 1 ms.

### CODE REVIEW

**Review by Marenz — DISMISSED** [review]

**Comment by leonardoalt** [comment]
Any ideas of how else we could test this?

**Comment by leonardoalt** [comment]
@Marenz ?

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `86406fb156dbb6b2750d35b5fe571a1fa621084f`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/model_checker_timeout_all/err:33:Note: 
 test/cmdlineTests/model_checker_timeout_all/err:50:Note: 
 test/cmdlineTests/model_checker_timeout_bmc/err:21:Note: 
 test/cmdlineTests/model_checker_timeout_bmc/err:38:Note: 

```
Please check that your changes are working as intended.

File: test/cmdlineTests/model_checker_timeout_bmc/input.sol [hunk 1]
```diff
@@ -2,9 +2,11 @@
...
-    function f(uint x) public pure {
-		assert(x > 0);
-		assert(x > 2);
-		assert(x > 4);
+	uint[][] a;
+    function f(uint x, uint n) public {
+		for (uint i = 0; i < n; ++i)
+			for (uint j = 0; j < n; ++j)
+				a[i][j] = i * j;
+		assert(a[3][4] == 12);
```
Comment (hrkrshnn) [inline test/cmdlineTests/model_checker_timeout_bmc/input.sol]: Why a loop for `CHC`?

File: test/cmdlineTests/model_checker_timeout_all/err [hunk 2]
```diff
@@ -1,50 +1,36 @@
-Warning: CHC: Assertion violation might happen here.
- --> model_checker_timeout_all/input.sol:6:3:
+Warning: CHC: Division by zero might happen here.
+ --> model_checker_timeout_all/input.sol:6:11:
   |
-6 | 		assert(x > 0);
-  | 		^^^^^^^^^^^^^
+6 | 		require(x % 7 == 0);
```
Comment (hrkrshnn) [inline test/cmdlineTests/model_checker_timeout_all/err]: How can division by zero happen here?

Comment (hrkrshnn) [reply test/cmdlineTests/model_checker_timeout_all/err]: Is this because of the timeout?

Comment (leonardoalt) [reply test/cmdlineTests/model_checker_timeout_all/err]: every mod and div expression checks for that

Comment (leonardoalt) [reply test/cmdlineTests/model_checker_timeout_all/err]: Yes, the query fails because of the timeout

Comment (leonardoalt) [reply test/cmdlineTests/model_checker_timeout_all/err]: that's why it says "might"

Comment (hrkrshnn) [reply test/cmdlineTests/model_checker_timeout_all/err]: So can't the test cause problems here, i.e., the solver can figure out that division by zero cannot happen here?

Would it be better to increase the time so that the division by zero error message disappears?

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `21b8fd67ee4748388d4478d7aaa4a27305beb673`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/model_checker_timeout_all/err:33:Note: 
 test/cmdlineTests/model_checker_timeout_all/err:50:Note: 
 test/cmdlineTests/model_checker_timeout_bmc/err:21:Note: 
 test/cmdlineTests/model_checker_timeout_bmc/err:38:Note: 

```
Please check that your changes are working as intended.

File: libsolidity/formal/BMC.cpp [hunk 3]
```diff
@@ -828,14 +828,14 @@ void BMC::checkCondition(
 	string extraComment = SMTEncoder::extraComment();
 	if (m_loopExecutionHappened)
 		extraComment +=
-			"\nNote that some information is erased after the execution of loops.\n"
-			"You can re-introduce information using require().";
+			"Note that some information is erased after the execution of loops.\n"
```
Comment (hrkrshnn) [inline libsolidity/formal/BMC.cpp]: Why these changes? The rest of the PR looks fine.

Comment (leonardoalt) [reply libsolidity/formal/BMC.cpp]: yea, just an attempt to make one of the other tests work. If the tests pass here I'll remove these

**Comment by leonardoalt** [comment]
Ready for review again @hrkrshnn @axic

**Review by hrkrshnn — APPROVED** [review]
Hopefully the tests will pass!

**Comment by leonardoalt** [comment]
@hrkrshnn they should. They passed yesterday with the second commit, which I extracted into a PR that was already merged.
